File: android/ui/components/UltraProExperienceTimeline.kt
package com.ultrapro.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.Card
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

data class UltraProExperienceItem(
    val role: String,
    val company: String,
    val period: String,
    val description: String
)

@Composable
fun UltraProExperienceTimeline(
    items: List<UltraProExperienceItem>,
    modifier: Modifier = Modifier
) {
    LazyColumn(
        modifier = modifier.fillMaxWidth(),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        items(items) { item ->
            Card(modifier = Modifier.fillMaxWidth()) {
                Column(modifier = Modifier.padding(20.dp)) {
                    Text(text = item.role, style = MaterialTheme.typography.titleLarge)
                    Text(
                        text = "${item.company} • ${item.period}",
                        style = MaterialTheme.typography.bodyMedium
                    )
                    Text(
                        text = item.description,
                        style = MaterialTheme.typography.bodyLarge,
                        modifier = Modifier.padding(top = 8.dp)
                    )
                }
            }
        }
    }
}

File: lib/ui/components/ultra_pro_experience_timeline.dart
import 'package:flutter/material.dart';

class UltraProExperienceItem {
  final String role;
  final String company;
  final String period;
  final String description;

  const UltraProExperienceItem({
    required this.role,
    required this.company,
    required this.period,
    required this.description,
  });
}

class UltraProExperienceTimeline extends StatelessWidget {
  final List<UltraProExperienceItem> items;

  const UltraProExperienceTimeline({
    super.key,
    required this.items,
  });

  @override
  Widget build(BuildContext context) {
    return ListView.separated(
      shrinkWrap: true,
      physics: const NeverScrollableScrollPhysics(),
      itemCount: items.length,
      separatorBuilder: (_, __) => const SizedBox(height: 16),
      itemBuilder: (context, index) {
        final item = items[index];
        return Card(
          child: Padding(
            padding: const EdgeInsets.all(20),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(item.role, style: Theme.of(context).textTheme.titleLarge),
                const SizedBox(height: 4),
                Text(
                  '${item.company} • ${item.period}',
                  style: Theme.of(context).textTheme.bodyMedium,
                ),
                const SizedBox(height: 8),
                Text(item.description, style: Theme.of(context).textTheme.bodyLarge),
              ],
            ),
          ),
        );
      },
    );
  }
}
